pkg/apis/networking/types.go

```diff
@@ -48,31 +48,35 @@ const (
 
 // NetworkPolicySpec provides the specification of a NetworkPolicy
 type NetworkPolicySpec struct {
-	// Selects the pods to which this NetworkPolicy object applies. The array of
-	// ingress rules is applied to any pods selected by this field. Multiple network
-	// policies can select the same set of pods. In this case, the ingress rules for
-	// each are combined additively. This field is NOT optional and follows standard
+	// Selects the pods to which this NetworkPolicy object applies for ingress and/or egress
+	// (as determined by `policyTypes`).  The relevant one or ones of `ingress` and `egress`
+	// list connections that this policy allows to be made from and to the selected pods.
```

Comment:
> The relevant one or ones of `ingress` and `egress` list connections that this policy allows to be made from and to the selected pods.

I find this a bit tricky to parse. What about something like this:

> ingress lists connections that this policy allows to be made to the selected pods, and egress lists connections that this policy allows to be made from the selected pods.

Reply:
Alternatively, maybe this detail should be specified in the `ingress` and `egress` fields rather than `podSelector`?

Reply:
I like your first revision.  I considered splitting most of this comment into the comments on the Ingress and Egress fields, but decided against that because this is all working towards explaining how selections combine.